### PR TITLE
Add "Check if TestCafe runs on page"

### PR DESCRIPTION
**Pull request by DIRECTcut**
#2 "How can my application code know whether the app is being tested with testcafe?"

### CODE REVIEW

File: examples/check-if-testcafe-active/README.MD [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Check Whether TestCafe Is Loaded
```
Comment (miherlosev) [inline examples/check-if-testcafe-active/README.MD:1]: `TestCafe is loaded` is a bad term. Better to use 'opened with TestCafe`. Here and below.

Check Whether Web Page Opened With TestCafe

File: examples/check-if-testcafe-active/README.MD [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+**Test Code**: [index.js](index.js)
+**Tested Page**: [index.html](index.html)
+
+If you want to change how your app operates (e.g. choose whether it should expose some data) depending on whether it is currently being tested with TestCafe. This example demonstrates how to let your application know whether TestCafe is loaded and running in the current window.
```
Comment (miherlosev) [inline examples/check-if-testcafe-active/README.MD:6]: You may need to change how your app operates depending on whether it is currently being tested with TestCafe.
This example demonstrates how to let your application know whether it is opened with TestCafe.

File: examples/check-if-testcafe-active/README.MD (hunk 1 above)
Comment (miherlosev) [inline examples/check-if-testcafe-active/README.MD:1]: `examples/check-if-testcafe-active/README.MD` -> `examples/check-if-opened-with-testcafe/README.MD`

File: examples/check-if-testcafe-active/index.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Check for TestCafe</title>
```
Comment (miherlosev) [inline examples/check-if-testcafe-active/index.html:6]: `<title>Check that the page opened with TestCafe</title>`

File: examples/check-if-testcafe-active/index.html [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+</head>
+<body>
```
Comment (miherlosev) [inline examples/check-if-testcafe-active/index.html:8]:
```js
<body>
     <div>
            The page is opened with TestCafe: <span id="result"></span>
     </div>
     <script>
            const resultSpan   = document.getElementById('result');
            const isInTestCafe = !!window['testCafeDriverInstance%'];
            
            resultSpan.textContent = isInTestCafe ? 'yes' : 'no';
     </script>
</body>
```

File: examples/check-if-testcafe-active/index.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { Selector } from 'testcafe';
+
+fixture `Check if TestCafe active`
```
Comment (miherlosev) [inline examples/check-if-testcafe-active/index.js:3]: fixture `Check that the page opened with TestCafe`

File: examples/check-if-testcafe-active/index.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+    .page('./index.html')
+
+test('Check if TestCafe loaded', async t => {
+    const answer = Selector('#answer');
```
Comment (miherlosev) [inline examples/check-if-testcafe-active/index.js:7]:
```js
await t.expect(Selector('div').innerText).eql('The page is opened with TestCafe: yes');
```

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: examples/check-if-testcafe-active/index.html (hunk 4 above)
Comment (DIRECTcut) [reply examples/check-if-testcafe-active/index.html:8]: Doesn't quite work- shows `no`, with testcafe or without. This seems to work though:

> const isInTestCafe = !!window.localStorage.nativeStorageKey.startsWith('hammerhead');

File: examples/check-if-opened-with-testcafe/index.html [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Check that the page was opened with TestCafe</title>
+</head>
+<body>
+    <div id="main">
+           The page is opened with TestCafe: <span id="result"></span>
+    </div>
+    <script>
+        const resultSpan   = document.getElementById('result');
+        const isInTestCafe = !!window['testCafeDriverInstance%'];
```
Comment (miherlosev) [inline examples/check-if-opened-with-testcafe/index.html:14]: `const isInTestCafe = !!window['%testCafeDriverInstance%'];`

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: examples/check-if-opened-with-testcafe/README.MD [hunk 8]
```diff
@@ -0,0 +1,20 @@
+# Check Whether Web Page Is Opened With TestCafe
+
+**Test Code**: [index.js](index.js)
+**Tested Page**: [index.html](index.html)
+
+You may need to change how your app operates depending on whether it is currently being tested with TestCafe. This example demonstrates how to let your application know whether it is opened with TestCafe.
+
+The test page includes a `<div>` and a `<button>`. When the button is pressed, a script determines whether the webpage has been opened with TestCafe. The `<div>` element then displays a corresponding message.
```
Comment (miherlosev) [inline examples/check-if-opened-with-testcafe/README.MD:8]: We changed the test logic. Rewrite it.

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: examples/check-if-opened-with-testcafe/README.MD [hunk 9]
```diff
@@ -0,0 +1,17 @@
+# Check Whether Web Page Is Opened With TestCafe
+
+**Test Code**: [index.js](index.js)
+**Tested Page**: [index.html](index.html)
+
+You may need to change how your app operates depending on whether it is currently being tested with TestCafe. This example demonstrates how to let your application know whether it is opened with TestCafe.
+
+The test page includes a `<div>`. When the page opens, a script determines whether the page has been opened with TestCafe. The `<div>` element then displays a corresponding message.
+
+During the test, the `t.expect.eql` method checks the answer text.
```
Comment (miherlosev) [inline examples/check-if-opened-with-testcafe/README.MD:10]: `checks the message text.`

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: examples/check-if-opened-with-testcafe/README.MD (hunk 9 above)
Comment (DIRECTcut) [reply examples/check-if-opened-with-testcafe/README.MD]: edited

**Review by miherlosev — APPROVED** [review]

File: examples/check-if-opened-with-testcafe/README.MD [hunk 10]
```diff
@@ -0,0 +1,17 @@
+# Check Whether Web Page Is Opened With TestCafe
+
+**Test Code**: [index.js](index.js)
+**Tested Page**: [index.html](index.html)
+
+You may need to change how your app operates depending on whether it is currently being tested with TestCafe. This example demonstrates how to let your application know whether it is opened with TestCafe.
```
Comment (VasilyStrelyaev) [inline examples/check-if-opened-with-testcafe/README.MD:6]: `change how your app operates` sounds like big changes, so it's a bit weird that someone has to make such changes for testing.

Maybe 

*You may need to make adjustments to your application depending on whether it is tested with TestCafe.*

File: examples/check-if-opened-with-testcafe/README.MD [hunk 11]
```diff
@@ -0,0 +1,17 @@
+# Check Whether Web Page Is Opened With TestCafe
```
Comment (VasilyStrelyaev) [inline examples/check-if-opened-with-testcafe/README.MD:1]: a very minor thing: I guess we shouldn't capitalize `Is` in titles, should we?

**Review by VasilyStrelyaev — CHANGES_REQUESTED** [review]

File: examples/check-if-opened-with-testcafe/README.MD (hunk 11 above)
Comment (DIRECTcut) [reply examples/check-if-opened-with-testcafe/README.MD:1]: I'm pretty sure we capitalize verbs. [Source](https://capitalizemytitle.com/#:~:text=The%20rules%20are%20fairly%20standard,%2C%20coordinating%20conjunctions%2C%20and%20prepositions.)

File: examples/check-if-opened-with-testcafe/README.MD (hunk 10 above)
Comment (DIRECTcut) [reply examples/check-if-opened-with-testcafe/README.MD:6]: I don't think that's quite true. The user wants their app to behave one way if it's launched through TestCafe, and the other if it isn't. There are no adjustments to the app itself- just to the environment. I suggest we leave this as is.

Or:

> You may need to adjust how your app functions depending on whether...

Comment (DIRECTcut) [reply examples/check-if-opened-with-testcafe/README.MD]: edited

File: examples/check-if-opened-with-testcafe/index.js [hunk 12]
```diff
@@ -0,0 +1,8 @@
+import { Selector } from 'testcafe';
+
+fixture `Check that the page opened with TestCafe`
```
Comment (arminal) [inline examples/check-if-opened-with-testcafe/index.js:3]:
```suggestion
fixture `Check to ensure that the page was opened with TestCafe`
```

File: examples/check-if-opened-with-testcafe/index.html [hunk 13]
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Check that the page was opened with TestCafe</title>
```
Comment (arminal) [inline examples/check-if-opened-with-testcafe/index.html:6]:
```suggestion
    <title>Check to ensure that the page was opened with TestCafe</title>
```

File: examples/check-if-opened-with-testcafe/index.html [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+</head>
+<body>
+    <div id="main">
+           The page is opened with TestCafe: <span id="result"></span>
```
Comment (arminal) [inline examples/check-if-opened-with-testcafe/index.html:10]:
```suggestion
           The page was opened with TestCafe: <span id="result"></span>
```

File: examples/check-if-opened-with-testcafe/index.js [hunk 15]
```diff
@@ -0,0 +1,8 @@
...
+    .page('./index.html')
+
+test('Check that the page opened with TestCafe', async t => {
```
Comment (arminal) [inline examples/check-if-opened-with-testcafe/index.js:6]:
```suggestion
test('Check to ensure that the page was opened with TestCafe', async t => {
```

File: examples/check-if-opened-with-testcafe/README.MD [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+
+**Test Code**: [index.js](index.js)
+**Tested Page**: [index.html](index.html)
+
+You may need to make adjustments to your application depending on whether it is tested with TestCafe. This example demonstrates how to let your application know whether it is opened with TestCafe.
+
+The test page includes a `<div>`. When the page opens, a script determines whether the page has been opened with TestCafe. The `<div>` element then displays a corresponding message.
```
Comment (arminal) [inline examples/check-if-opened-with-testcafe/README.MD:8]:
```suggestion
The test page includes a `<div>`. When the page opens, a script determines whether the page has been opened with TestCafe. The `<div>` element then displays the corresponding message.
```

File: examples/check-if-opened-with-testcafe/index.html (hunk 13 above)
Comment (DIRECTcut) [reply examples/check-if-opened-with-testcafe/index.html:6]: > Check whether the page was opened with TestCafe

File: examples/check-if-opened-with-testcafe/index.js (hunk 12 above)
Comment (DIRECTcut) [reply examples/check-if-opened-with-testcafe/index.js:3]: > Check whether the page was opened with TestCafe

File: examples/check-if-opened-with-testcafe/index.js (hunk 15 above)
Comment (DIRECTcut) [reply examples/check-if-opened-with-testcafe/index.js:6]: > Check whether the page was opened with TestCafe

**Review by VasilyStrelyaev — APPROVED** [review]

File: examples/check-if-opened-with-testcafe/README.MD [hunk 17]
```diff
@@ -0,0 +1,17 @@
+# Check Whether Web Page Is Opened With TestCafe
+
+**Test Code**: [index.js](index.js)
+**Tested Page**: [index.html](index.html)
+
+You may need to make adjustments to your application depending on whether it is tested with TestCafe. This example demonstrates how to let your application know whether it is opened with TestCafe.
```
Comment (dirk-pieterse) [inline examples/check-if-opened-with-testcafe/README.MD:6]: "You may need to make adjustments to your application depending on whether it is tested with TestCafe" This is an 'empty' sentence, and by that I mean it lacks real meaning. For example, "depending on whether it is tested with TestCafe": this topic is about TestCafe, so why would it depend on this? "You may need to make adjustments": which 'adjustments'? Change options? To be honest, it is difficult to tell what you are trying to say here. Do you need to change your application before you can test it in TestCafe? Why do you need to 'let your application know' that you opened it in TestCafe?

File: examples/check-if-opened-with-testcafe/README.MD [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+
+The test page includes a `<div>`. When the page opens, a script determines whether the page has been opened with TestCafe. The `<div>` element then displays the corresponding message.
```
Comment (dirk-pieterse) [inline examples/check-if-opened-with-testcafe/README.MD:8]:
```suggestion
The test page includes a `<div>`. When the page opens, a script indicates that the page was opened with TestCafe. The `<div>` element then displays the corresponding message.
```

Comment (dirk-pieterse) [reply examples/check-if-opened-with-testcafe/README.MD:8]: "displays the corresponding message" Which corresponding message?

File: examples/check-if-opened-with-testcafe/README.MD [hunk 19]
```diff
@@ -0,0 +1,17 @@
...
+
+During the test, the `t.expect.eql` method checks the message text.
```
Comment (dirk-pieterse) [inline examples/check-if-opened-with-testcafe/README.MD:10]: "checks the message text" Checks it for what? What is it looking for?

File: examples/check-if-opened-with-testcafe/README.MD (hunk 11 above)
Comment (dirk-pieterse) [reply examples/check-if-opened-with-testcafe/README.MD:1]:
```suggestion
# Check whether the web page is opened with TestCafe
```

File: examples/check-if-opened-with-testcafe/README.MD [hunk 20]
```diff
@@ -0,0 +1,17 @@
...
+
+## TestCafe Functions and Methods Employed in This Example
```
Comment (dirk-pieterse) [inline examples/check-if-opened-with-testcafe/README.MD:12]:
```suggestion
## TestCafe Functions and Methods used in this example
```

**Review by dirk-pieterse — APPROVED** [review]

File: examples/check-if-opened-with-testcafe/README.MD [hunk 21]
```diff
@@ -0,0 +1,19 @@
+# Check Whether Web Page Is Opened With TestCafe
+
+**Test Code**: [index.js](index.js)
+**Tested Page**: [index.html](index.html)
+
+When TestCafe runs, it exposes multiple variables with internal TestCafe data to the global object. Your application can check if any of those variables are present and conclude whether it is opened with TestCafe.
```
Comment (dirk-pieterse) [inline examples/check-if-opened-with-testcafe/README.MD:6]:
```suggestion
When TestCafe runs, it exposes multiple variables with internal TestCafe data to the global object. Your application can check if these variables are present to determine whether it is opened with TestCafe.
```

File: examples/check-if-opened-with-testcafe/README.MD [hunk 22]
```diff
@@ -0,0 +1,19 @@
...
+
+This example demonstrates how to let your application know whether it is opened with TestCafe.
+
+The test page includes a `<div>` with a `<span>`. When the page is opened, a script determines whether the page was opened with TestCafe. The `<span>` element then displays `yes` or `no` accordingly.
```
Comment (dirk-pieterse) [inline examples/check-if-opened-with-testcafe/README.MD:10]:
```suggestion
The test page includes a `<div>` with a `<span>`. When the page is opened, a script determines whether the page was opened with TestCafe. The `<span>` element then displays `yes` or `no`.
```

**Review by dirk-pieterse — APPROVED** [review]

**Pull request merged**